src/battle_ai_util.c

```diff
@@ -2486,6 +2486,18 @@ u32 GetIndexInMoveArray(u32 battler, u32 move)
     return MAX_MON_MOVES;
 }
 
+u32 GetBattlerMoveIndexWithEffect(u32 battler, enum BattleMoveEffects effect)
+{
+    u16 *moves = GetMovesArray(battler);
+    u32 i;
+    for (i = 0; i < MAX_MON_MOVES; i++)
```

Comment:
```suggestion
    for (u32 i = 0; i < MAX_MON_MOVES; i++)
```
Loop iterator declared outside of loop declaration